crates/oxide/src/extractor/mod.rs

```diff
@@ -139,6 +140,41 @@ impl<'a> Extractor<'a> {
 
         extracted
     }
+
+    pub fn extract_variables_from_css(&mut self) -> Vec<Extracted<'a>> {
```

Comment:
Could this be done in a pre-processor where we already have logic to do that based on file extension?

Reply:
Sort of but not really, we do have the contents and the extension so we do have all the information we need, but we return a new potentially transformed input file.

```rs
pub fn pre_process_input(content: &[u8], extension: &str) -> Vec<u8> { … }
```

But we don't emit extracted variables here. So we could update this logic to also emit `Vec<Extracted>`. 

Another solution is that we could transform the CSS file and replace everything that is _not_ a used CSS variable with whitespace. That way the normal extractor won't find anything, and the CSS extractor will only find used variables.

But both these solutions seem like a hack, and a bit of an abuse for this `pre_process_input` function 😅 unless you're thinking about something completely different?

Reply:
Yeah I think this does make sense, agree that abusing pre-process for this does feel wrong too now that I think about it more.